src/gsy_e/models/market/forward.py

```diff
@@ -58,17 +69,50 @@ def _get_start_time(current_time: DateTime) -> DateTime:
     def _get_end_time(current_time: DateTime) -> DateTime:
         """Return time when the market block ends."""
 
+    @property
+    @abstractmethod
+    def _time_from_market_close_till_delivery(self) -> duration:
+        """
+        Retrieves the time duration from the time that the market closes till the time that the
+        traded energy should be delivered.
+        """
+
     @staticmethod
     @abstractmethod
     def _get_market_slot_duration(current_time: DateTime, _config) -> duration:
         """Return duration of market slots inside the market block."""
 
     def create_future_market_slots(self, current_market_time_slot: DateTime,
-                                   config: "SimulationConfig") -> None:
+                                   config: "SimulationConfig") -> List[DateTime]:
         if not ConstSettings.ForwardMarketSettings.ENABLE_FORWARD_MARKETS:
-            return
-        self._create_future_market_slots(self._get_start_time(current_market_time_slot),
-                                         self._get_end_time(current_market_time_slot), config)
+            return []
+        created_future_slots = self._create_future_market_slots(
+            self._get_start_time(current_market_time_slot),
+            self._get_end_time(current_market_time_slot), config)
+
+        self._set_open_market_slot_parameters(current_market_time_slot, created_future_slots)
+
+        return created_future_slots
+
+    def get_market_parameters_for_market_slot(
+            self, market_slot: DateTime) -> ForwardMarketSlotParameters:
+        """Retrieve the parameters for the selected market slot."""
+        return self._open_market_slot_parameters.get(market_slot)
+
+    def _set_open_market_slot_parameters(
+            self, current_market_slot: DateTime, created_market_slots: List[DateTime]):
+        """Update the parameters of the newly opened market slots."""
+        for market_slot in created_market_slots:
+            if market_slot in self._open_market_slot_parameters:
+                continue
+
+            self._open_market_slot_parameters[market_slot] = ForwardMarketSlotParameters(
+                delivery_start_timestamp=market_slot,
+                delivery_end_timestamp=(
+                        market_slot + self._get_market_slot_duration(market_slot, None)),
+                open_timestamp=current_market_slot,
+                close_timestamp=market_slot - self._time_from_market_close_till_delivery
```

Comment:
Shouldn't this just be the same as `_get_end_time()`?

Reply:
As I mentioned in the previous comment, this is not the closing time of the market with a specific market slot, but rather the time that the rotator should stop creating new markets in the future, based on the current time slot.